client/my-sites/checkout/composite-checkout/hooks/product-variants.tsx

```diff
@@ -203,7 +203,7 @@ function isVariantAllowed(
 }
 
 function VariantPrice( { variant }: { variant: AvailableProductVariantAndCompared } ) {
-	const currentPrice = variant.priceFinal || variant.priceFull;
+	const currentPrice = variant.introductoryOfferPrice ?? variant.priceFinal ?? variant.priceFull;
```

Comment:
I think I understand the intent here, but I'm a little confused about the data. This seems to be saying, "use the introductoryOfferPrice, but if that's not set, use the priceFinal, and if that's not set, use priceFull". However, it does this with nullish coalescing, and as far as the types say, `priceFinal` will never be `null` or `undefined`. So while the previous version of this line would display `priceFull` if `priceFinal` was `0`, the new version of this line is guaranteed to never display `priceFull` (unless the types are wrong). This seems like possibly a bug.

Reply:
> use the introductoryOfferPrice, but if that's not set, use the priceFinal, and if that's not set, use priceFull

Is definitely the intent. I think I got excited with the null coalescing operator. Since introductoryOfferPrice can be null ( in which case we use the original logic ) or 0 ( which is possible for free trial introductory offers ) we need something more like.

```js
const currentPrice =
		variant.introductoryOfferPrice !== null
			? variant.introductoryOfferPrice
			: variant.priceFinal || variant.priceFull;
```